packages/core/src/storage/didcomm/DidCommMessageRecord.ts

```diff
@@ -81,8 +77,6 @@ export class DidCommMessageRecord extends BaseRecord<DefaultDidCommMessageTags>
       threadId,
       protocolName,
       messageName,
-      versionMajor,
-      versionMinor,
```

Comment:
I've maybe been to quick to remove them.

The default way for indy to store array values is by storing the value in the key and setting
the tag value to "1".

So when we store a message for a protocol version of 1.1 it will convert it to array when retrieving from storage.

Currently we use the exact '@type' of a message to retrieve it. If we're going to support more versions of protocols I thought we could retrieve the message by protocol name and message name and do the version filtering post.

However it may be useful to retrieve e.g. all 1.x messages in the future.

Will think about it for a bit, but we would have to store the value of '1' as something different so we don't transform it into an array, or check if the key doesn't contain an :

Reply:
Looking at it, it's not arrays but booleans. We check if the value === '1' and base the boolean of off that.

```
      } else if (value === '1' || value === '0') {
        transformedTags[key] = value === '1'
      }
```

What we could do is store the number variant as `__1` or something so we know for sure it's the number 1 (or 0) and not the boolean value. Not sure what is best here. Thoughts?

Reply:
Sorry, I likely am missing something--I'll admit I'm not quite sure what the issue is. If we store a tag with a key of 'versionMajor' with a value of '1', shouldn't that be fine? What am I missing?

Reply:
Indy only allows tag values to be strings, while in AFJ tag values can be string, number, boolean and array of those types. So we have custom processing to go from afj tag to indy tag and vice versa

Boolean values are transformed by us to the string value '1'. So if we're storing the actual value 1 it will also be stored as string value '1'. When retrieved from storage afj will transform the value 1 to a boolean value.